src/interfaces/simulation.ts

```diff
@@ -500,16 +452,134 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     return !!this.yearn.services.partner?.isAllowed(vault);
   }
 
-  private async simulateZapApprovalTransaction(
-    zapApprovalTransaction: ZapApprovalTransactionOutput,
-    options: SimulationOptions
-  ): Promise<SimulationResponse> {
-    options.save = true;
-    return await this.simulationExecutor.makeSimulationRequest(
-      zapApprovalTransaction.from,
-      zapApprovalTransaction.to,
-      zapApprovalTransaction.data,
-      options
+  private async getZapperZapInApprovalData({
+    sellToken,
+    from,
+    toVault,
+    options,
+  }: DepositProps): Promise<ZappingApprovalData> {
+    if (!isEthereum(this.chainId)) {
```

Comment:
what about having this on zaps?

Reply:
What do you mean by "on zaps"?

Reply:
Oh, the new file that you added that seems to be a lot of utils around zapper

Reply:
Ah I see, we're still refactoring, so this might change. I think the idea is to move most of this logic out of the `simulation.ts` and this class would just call the simulation executor with the populated transaction from the vaults interface for example. I'll keep this suggestion in mind. Let's leave it unresolved for now so we don't forget

Reply:
@FiboApe checked that we do the chain check before in both of these cases https://github.com/yearn/yearn-sdk/pull/280/commits/3ac9a6a76ff2f84f8fabf7b430438e12e367fbea -- so they were removed.